packages/bot-web-ui/src/stores/journal-store.js

```diff
@@ -29,8 +36,14 @@ export default class JournalStore {
         { id: message_types.SUCCESS, label: localize('System') },
     ];
 
+    getJournalStorageKey = key => {
+        return JSON.parse(sessionStorage.getItem(key));
+    };
+
     @observable is_filter_dialog_visible = false;
-    @observable unfiltered_messages = [];
+    @observable unfiltered_messages = Array.isArray(this.getJournalStorageKey(journal_storage_key))
+        ? this.getJournalStorageKey(journal_storage_key)
+        : [];
```

Comment:
Instead of calling `JSON.parse(sessionStorage.getItem(key)` twice directly, you now call it twice indirectly. Perhaps you could assign it as below:

```JS
@observable unfiltered_messages = this.getJournalStorageKey(journal_storage_key) ?? [];
```

Reply:
`slice(0, 0)` selects the element starting at 0, and ends at 0, and does not include 0.
`slice()` returns the old array as a new array object.
Can be tried here as well: https://www.w3schools.com/jsref/tryit.asp?filename=tryjsref_slice_array
(here I wanted to clear the array and update it, on the other call I wanted to only update the array).